daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -120,5 +120,10 @@ func (w *RotateFileWriter) NotifyRotateEvict(sub chan interface{}) {
 
 // Close closes underlying file and signals all readers to stop.
 func (w *RotateFileWriter) Close() error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if w.f == nil {
```

Comment:
`w.f` is not really ever set to nil or could potentially be nil for other reasons.
I'd track his with a `closed` bool on `w`